src/NBitcoin.Tests/NetworkTests.cs

```diff
@@ -76,6 +66,10 @@ public void BitcoinMainnetIsInitializedCorrectly()
         {
             Network network = Network.Main;
 
+            Assert.Equal(15, network.Checkpoints.Count);
```

Comment:
I'm not sure if assert.Equal is the right approach here. 
In case we add a new valid checkpoint the test will fail. 

I guess the purpose of the test is to make sure that checkpoints were loaded correctly and none were removed so maybe here and in another similar asserts it will be a good idea to check if the number of checkpoints is less than what we have now? If it's greater- new checkpoint was added and we shouldn't fail the test.

Reply:
yes I want the test to fail if you update checkpoints. Basically I want to make sure that you've 1. loaded checkpoints 2. loaded the right set of checkpoints.

Reply:
Ok, makes sense. 

Just feels slightly weird that you can do a good change and it will fail the test.

Reply:
Not really: I'm just testing every property in the Network so any change made to it should also update the tests.

Reply:
To me we don't even need a test around checkpoints, its metadata of a network.
If someone is changing that its intentional and the test will obviously fail.
Same goes to other metadata of network (magic bytes, port)